airflow/config_templates/default_airflow.cfg

```diff
@@ -881,6 +881,15 @@ logs_volume_subpath =
 # A shared volume claim for the logs
 logs_volume_claim =
 
+# Extra volumes to be mounted in worker pods.  Volumes are specified as
+# keys in a JSON object with nested JSON values specifying each volume's
+# options.  Recognized options are `claim_name` or `secret_name`
+# (required), `mount_path` (required), `read_only` (boolean, default
+# null), `sub_path` (default null), `secret_key` (string, default null),
+# `secret_mode` (string, default null).
+# Example: extra_volume_mounts = "{{{{\"secret_vol\": {{{{\"secret_name\": \"some-secret\", \"mount_path\": \"/dir1\", \"sub_path\": \"subpath1\", \"secret_mode\": \"440\"}}}}, \"pvc\": {{{{\"claim_name\": \"some-pvc\", \"mount_path\": \"/dir2\"}}}}}}}}"
```

Comment:
Does a user have to do all of these escapes or can they just supply a dict? I'd rather not have people supplying strings this gnarly if possible.

Reply:
To the user, the value in the `.cfg` file needs to be valid JSON/a `dict`, but, yeah, it's very gnarly here.

Reply:
Do you want me to add something specific to the documentation before resolving this?

Reply:
@kaxil what do you think? should we add this to the airflow_local_settings.py so users can define python dicts?

Reply:
airflow_local_settings.py file is intended only for logs currently. I am not sure if we want to mix it.

Reply:
@mik-laj we use the airflow_local_settings.py for the pod_mutation_hook https://github.com/apache/airflow/blob/8465d66f05baeb73dd4479b019515c069444616e/airflow/settings.py#L107. That's why I was thinking it might make sense.

Reply:
The dict in Example looks ugly but that is a separate issue, I would solve it this week so we won't need to define anything in airflow_local_settings.py . `extra_volume_mounts` feels like a candidate for airflow.cfg similar to kubernetes_secrets etc.

@dimberman It might make sense to convert this `extra_volume_mounts` to a section similar to kubernetes_secrets, thoughts ?